pkg/option/config.go

```diff
@@ -2346,8 +2353,16 @@ func (c *DaemonConfig) Populate() {
 
 	c.populateDevices()
 
-	if nativeCIDR := viper.GetString(IPv4NativeRoutingCIDR); nativeCIDR != "" {
-		c.ipv4NativeRoutingCIDR = cidr.MustParseCIDR(nativeCIDR)
+	if c.IPAM == ipamOption.IPAMOperator {
+		var nativeCIDR string
+		c.IPAMClusterPoolIPv4RoutingCIDR = viper.GetStringSlice(operatorOption.IPAMOperatorV4CIDR)
+		if nativeCIDR = strings.Join(c.IPAMClusterPoolIPv4RoutingCIDR[:0], ""); nativeCIDR != "" {
+			c.ipv4NativeRoutingCIDR = cidr.MustParseCIDR(nativeCIDR)
+		}
+	} else {
+		if nativeCIDR := viper.GetString(IPv4NativeRoutingCIDR); nativeCIDR != "" {
+			c.ipv4NativeRoutingCIDR = cidr.MustParseCIDR(nativeCIDR)
+		}
```

Comment:
We should do this regardless of IPAM being the operator or not. The user might have other CIDRs in mind to set in the `ipv4NativeRoutingCIDR` so we should not overwrite it as we are doing in 2360.

Reply:
Ok I will address your comments.

Reply:
addressed review comments.